### ISSUE TITLE
md5sum is broken due to file open mode issue

### ISSUE THREAD

**Issue by iquadtree** [rmyorston/busybox-w32#15]
Reproducable with any particular binary file. Just compare the output of busybox md5sum and another MD5 hash calculator like md5sum from Cygwin. Looks like the same issue as #4.

I figured out that _open() CRT call always uses text file mode (_O_TEXT) by default when opening files until it will set explicitly by the _set_fmode(). Thus ^Z (0x1A) is treated like end-of-file sequence and reading is stopped.

I've tried to fix it with following patch:

```diff
@@ -125,10 +125,12 @@ int mingw_open (const char *filename, int oflags, ...)
               va_start(args, oflags);
              mode = va_arg(args, int);
              va_end(args);
+             oflags |= _O_BINARY; // force binary mode by default
+
               if (oflags & O_NONBLOCK) {
                             errno = ENOSYS;
                             return -1;
              }
              if (filename && !strcmp(filename, "/dev/null"))
```
... and it worked for me. I'm not sure about regression risks (please take a look at #4) but I suppose that line endings issue can be solved by implementing -a command-line switch for diff which seems to be ignored.

**Comment by rmyorston** [comment]
I think this has been fixed by commit f9bba0a5.  As well as setting `_fmode` when building with mingw-w64 this commit also explicitly sets the modes of the `stdin`, `stdout` and `stderr` streams when starting an applet.

**Comment by iquadtree** [comment]
Sounds good! Thank you.
